sdk/iot/hub/samples/telemetry/src/iot_hub_telemetry_example.c

```diff
@@ -0,0 +1,278 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifdef _MSC_VER
+// warning C4201: nonstandard extension used: nameless struct/union
+#pragma warning(disable : 4201)
+#endif
+#include <paho-mqtt/MQTTClient.h>
+#ifdef _MSC_VER
+#pragma warning(default : 4201)
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#ifdef _WIN32
+// Required for Sleep(DWORD)
+#include <Windows.h>
+#else
+// Required for sleep(unsigned int)
+#include <unistd.h>
+#endif
+
+#include <az_iot_hub_client.h>
+#include <az_result.h>
+#include <az_span.h>
+
+// TODO: #564 - Remove the use of the _az_cfh.h header in samples.
+//              Note: this is required to work-around MQTTClient.h as well as az_span init issues.
+#include <_az_cfg.h>
+
+// Device ID
+#define DEVICE_ID "AZ_IOT_DEVICE_ID"
+
+// IoT Hub FQDN
+#define IOT_HUB_FQDN "AZ_IOT_HUB_FQDN"
+
+// Device information
+#define REGISTRATION_ID_ENV "AZ_IOT_REGISTRATION_ID"
```

Comment:
how about checking if user has passed these macro using compiler option use that,  otherwise use default in c file?

ifndef DEVICE_ID
#define DEVICE_ID ""
#endif

Reply:
Having security or privacy related information within C files is not a good idea. We shouldn't promote that practice.

For larger embedded devices the configuration should be either read from a file, environment or, ideally from a Hardware Security Module.
For smaller devices a separate flash segment, eeprom or HSM should be used.

Reply:
These macro's aren't the values but the environment variables names to look for. I believe it is settled that anything user specific for samples will be done via environment variables